src/legacy/core_plugins/kibana/public/dashboard/np_ready/dashboard_app_controller.tsx

```diff
@@ -713,8 +715,44 @@ export class DashboardAppController {
         });
     }
 
+    const hasUrlParam = (param: string): boolean =>
+      param in parse(location.hash.slice(1), true).query;
```

Comment:
We already have a mechanism for checking URL parameters. Can you make use of `$routeParams` here? Eg:
https://github.com/elastic/kibana/blob/master/src/legacy/core_plugins/kibana/public/dashboard/np_ready/dashboard_app_controller.tsx#L354